evelynnchen-cmu/translator-service
# Write a Mock test

# Add Mock Tests for Translator Service

## Description
This PR adds a comprehensive test suite for the translator service, implementing unit tests for both the Flask application endpoints and the translator functionality. The tests use pytest and unittest.mock to ensure proper functionality without making actual API calls to Azure OpenAI.

## Changes Made
- Added mock tests for the translator module covering:
  - Empty message handling
  - Gibberish text detection
  - English text processing
  - Non-English text translation
  - Error handling and edge cases
  - Invalid API responses

- Added Flask application endpoint tests covering:
  - Empty query parameter handling
  - Valid query processing
  - Complete translation flow

## Test Coverage
The test suite covers the following scenarios:
- Basic functionality:
  - Empty message validation
  - Gibberish detection
  - Language detection
  - Translation processing
- Error handling:
  - API errors
  - Unexpected response formats
  - Edge cases
- API endpoint testing:
  - Query parameter handling
  - Response format validation
  - Integration with translator module

## Testing Instructions
1. Ensure you have the required packages:
```bash
pip install pytest pytest-mock
